Bottom: d104226c9c1588bc14a7266dfd5f014f80b89aae
Top:    c98ffce4c7cf23fcebae20bb5f4eb872fa1354e0
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-16 13:02:55 -0400

Refresh of mpc5200-ac97

---

diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index b481eca..13b53c3 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -1,5 +1,5 @@
 /*
- * linux/sound/mpc52xx-ac97.c -- AC97 support for the Freescale MPC52xx chip.
+ * linux/sound/mpc5200-ac97.c -- AC97 support for the Freescale MPC52xx chip.
  *
  * Copyright 2008 Jon Smirl, Digispeaker
  * Author: Jon Smirl <jonsmirl@gmail.com>
@@ -32,6 +32,7 @@
 #include <asm/mpc52xx_psc.h>
 
 #include "mpc5200_dma.h"
+#include "mpc5200_psc_ac97.h"
 
 MODULE_AUTHOR("Jon Smirl");
 MODULE_DESCRIPTION("MPC52xx AC97 module");
@@ -114,7 +115,7 @@ struct psc_ac97 {
 	} stats;
 };
 
-#define DRV_NAME "mpc52xx-psc-ac97"
+#define DRV_NAME "mpc5200-psc-ac97"
 
 static unsigned short psc_ac97_read(struct snd_ac97 *ac97, unsigned short reg)
 {
@@ -274,21 +275,21 @@ static struct snd_soc_dai_ops psc_ac97_digital_ops = {
 
 static struct snd_soc_dai psc_dai_ac97[] = {
 {
-	.name	= "mpc52xx AC97 analog",
-	.id	= 0,
+	.name	= "mpc5200 AC97 analog",
+	.id	= MPC5200_AC97_ANALOG,
 	.ac97_control	= 1,
 	.suspend = psc_ac97_suspend,
 	.resume = psc_ac97_resume,
 
 	.playback = {
-		.stream_name	= "mpc52xx AC97 analog",
+		.stream_name	= "mpc5200 AC97 analog",
 		.channels_min	= 1,
 		.channels_max	= 6,
 		.rates		= SNDRV_PCM_RATE_8000_48000,
 		.formats	= SNDRV_PCM_FORMAT_S32_BE,
 	},
 	.capture = {
-		.stream_name	= "mpc52xx AC97 analog",
+		.stream_name	= "mpc5200 AC97 analog",
 		.channels_min	= 1,
 		.channels_max	= 2,
 		.rates		= SNDRV_PCM_RATE_8000_48000,
@@ -297,13 +298,13 @@ static struct snd_soc_dai psc_dai_ac97[] = {
 	.ops 	= &psc_ac97_analog_ops,
 },
 {
-	.name	= "mpc52xx AC97 digital",
-	.id	= 1,
+	.name	= "mpc5200 AC97 digital",
+	.id	= MPC5200_AC97_DIGITAL,
 	.suspend = psc_ac97_suspend,
 	.resume = psc_ac97_resume,
 
 	.playback = {
-		.stream_name	= "mpc52xx AC97 digital",
+		.stream_name	= "mpc5200 AC97 digital",
 		.channels_min	= 1,
 		.channels_max	= 2,
 		.rates		= SNDRV_PCM_RATE_32000 | \
diff --git a/sound/soc/fsl/mpc5200_psc_ac97.h b/sound/soc/fsl/mpc5200_psc_ac97.h
index 0e0a84e..d028c55 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.h
+++ b/sound/soc/fsl/mpc5200_psc_ac97.h
@@ -1,13 +1,13 @@
 /*
- * Freescale MPC5200 PSC in I2S mode
+ * Freescale MPC5200 PSC in AC97 mode
  * ALSA SoC Digital Audio Interface (DAI) driver
  *
  */
- 
-#ifndef __SOUND_SOC_FSL_MPC52xx_PSC_I2S_H__
-#define __SOUND_SOC_FSL_MPC52xx_PSC_I2S_H__
 
-#define MPC52xx_CLK_INTERNAL 0
-#define MPC52xx_CLK_CELLSLAVE 1
+#ifndef __SOUND_SOC_FSL_MPC52xx_PSC_AC97_H__
+#define __SOUND_SOC_FSL_MPC52xx_PSC_AC97_H__
 
-#endif /* __SOUND_SOC_FSL_MPC52xx_PSC_I2S_H__ */
+#define MPC5200_AC97_ANALOG 0
+#define MPC5200_AC97_DIGITAL 1
+
+#endif /* __SOUND_SOC_FSL_MPC52xx_PSC_AC97_H__ */
